Header.c

```diff
@@ -198,8 +198,8 @@ void Header_draw(const Header* this) {
    for (int y = 0; y < height; y++) {
       mvhline(y, 0, ' ', COLS);
    }
-   const int numCols = HeaderLayout_getColumns(this->headerLayout);
-   const int width = COLS - 2 * pad - (numCols - 1);
+   const size_t numCols = HeaderLayout_getColumns(this->headerLayout);
+   const int width = COLS - 2 * pad - ((int)numCols - 1);
```

Comment:
I don't like this. If `numCols` needs to be downcast, then `numCols` should be an `unsigned int` type. Using `size_t` would be overkill.

Reply:
numCols was previously being downcast implicitly (HeaderLayout_getColumns returns size_t), now it is explicit.  The use of an explicit cast here now is to limit the scope of changes to a small, manageable set.  I expect many more things will transition to size_t, ideally also in small manageable increments, and these explicit casts will be slowly dropped out of the code over time.

Reply:
@natoscott I previously suggested that `HeaderLayout_getColumns` should return `int` or `unsigned int` in order to align with the type limit of terminal columns (`COLS`). `size_t` brings no advantage here. If the number of columns in HeaderLayout is greater than `COLS` then some columns will never be displayed. In other words it makes no sense to allocate so many columns for the Header UI.